examples/text-generation/textgen_pipeline/pipeline.py

```diff
@@ -0,0 +1,56 @@
+import os
+import sys
+
+import torch
+from transformers import TextGenerationPipeline
+
+
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir)))
+try:
+    from utils import initialize_model
+except:
+    raise
+
+
+class GaudiTextGenerationPipeline(TextGenerationPipeline):
+    def __init__(self, args, logger):
+        self.model, self.tokenizer, self.generation_config = initialize_model(args, logger)
+
+        self.device = args.device
+
+        if args.do_sample:
+            self.generation_config.temperature = args.temperature
+            self.generation_config.top_p = args.top_p
+
+        self.max_padding_length = args.max_input_tokens if args.max_input_tokens > 0 else 100
```

Comment:
is self.max_padding_length meant to be longest expected input sentence or number of tokens to be generated?

Reply:
Yes, it is a limit on the number of input tokens.